src/sentry/api/endpoints/project_transaction_threshold.py

```diff
@@ -90,29 +89,15 @@ def post(self, request: Request, project) -> Response:
 
         data = serializer.validated_data
 
-        with transaction.atomic(router.db_for_write(ProjectTransactionThreshold)):
-            try:
-                project_threshold = ProjectTransactionThreshold.objects.get(
-                    project=project,
-                    organization=project.organization,
-                )
-                project_threshold.threshold = data.get("threshold") or project_threshold.threshold
-                project_threshold.metric = data.get("metric") or project_threshold.metric
-                project_threshold.edited_by_id = request.user.id
-                project_threshold.save()
-
-                created = False
-
-            except ProjectTransactionThreshold.DoesNotExist:
-                project_threshold = ProjectTransactionThreshold.objects.create(
-                    project=project,
-                    organization=project.organization,
-                    threshold=data.get("threshold", 300),
-                    metric=data.get("metric", TransactionMetric.DURATION.value),
-                    edited_by_id=request.user.id,
-                )
-
-                created = True
+        project_threshold, created = ProjectTransactionThreshold.objects.update_or_create(
+            project=project,
+            organization=project.organization,
+            defaults={
+                "threshold": data.get("threshold", 300),
+                "metric": data.get("metric", TransactionMetric.DURATION.value),
```

Comment:
This looks like it will update the row with `TransactionMetric.Duration.value` if the metric is not specified. I think what you want is the to pass the metric to `defaults` if it's given and pass `TransactionMetric.Duration.value` to `create_defaults` if it's not given.

A test to make sure this has the right behaviour when updating a threshold without specifying a metric where the metric was previously changed to something like LCP would be nice.

Reply:
oof good catch